core/src/log/elastic.rs

```diff
@@ -0,0 +1,188 @@
+use chrono::prelude::{SecondsFormat, Utc};
+use itertools;
+use reqwest;
+use serde::ser::Serializer as SerdeSerializer;
+use std::fmt;
+use std::fmt::Write;
+use std::result::Result as StdResult;
+
+use graph::slog::*;
+use graph::slog_async;
+
+/// General configuration parameters for Elasticsearch logging.
+#[derive(Clone, Debug)]
+pub struct ElasticLoggingConfig {
+    /// The Elasticsearch service to log to.
+    pub endpoint: String,
+    /// The Elasticsearch username.
+    pub username: String,
+    /// The Elasticsearch password (optional).
+    pub password: Option<String>,
```

Comment:
In the ES API, what does an empty user or password mean? If the username is really optional, then it should be an `Option` here, you can do `unwrap_or("")` when calling `basic_auth`.